### ISSUE TITLE
Feature Request: tiled windows within stacked?

### ISSUE THREAD

**Issue by peey** [pop-os/shell#626]
I understand that the current tiled stacking mode lets you create stacks within a tile.

e.g. if X|Y is X and Y tiled and X/Y is X and Y stacked, then you can have A/C tiled with B/D (2 windows on the left, 2 on the right)

But you can't have A|B stacked with C|D

I'd previously opened up a related issue at an earlier time but was unsure if it really added value: https://github.com/pop-os/shell/issues/273

But I think it does. Essentially, you might want to have two window groups on the same workspace with different tiling layouts. In pop-shell, if you can "stack" these window groups, then that'd be great.

I think the limitation comes from pop-shell having only one top level fork for tiling, and it has a special designation. Another issue might be that you can only stack single windows, and not abstract window groups. 

If it was possible to have a fork anywhere (e.g. even within a stack), then that may make this possible.

I'm now speculating, but if instead of "fork"s here https://github.com/pop-os/shell#how-it-works there was a generic "window group" which could either be tiled (behaving similar to the current fork) or be "stacked" then I think that might simplify the data structures, and also perhaps the mental model (e.g. I could quickly turn any fork into a stack with a single command, and perhaps also the other way around)

**Comment by mmstick** [comment]
Stacks are technically window groups. Are you asking for stacks to support holding forks in addition to windows?

**Comment by peey** [comment]
> Are you asking for stacks to support holding forks in addition to windows?

Yes!

**Comment by timothyandrew** [comment]
+1 here! The ability to tile terminal windows within a stack would be extremely useful.

**Comment by ezhang7423** [comment]
I agree, would love this feature. Would this get merged if I built it out?

**Comment by jacobgkau** [comment]
> Would this get merged if I built it out?

I don't see why it wouldn't be, although it may require UX review to make sure it's not too confusing (and will of course also require QA review to make sure it doesn't break anything else.)
